### DISCUSSION TITLE
SQL Query use table without "Default_" prefix

### DISCUSSION THREAD

**Discussion by alXDraghiciu** [OrchardCMS/OrchardCore#12785] [Q&A]
Hi. I'm trying to create a SQL Query to read data from a table in the database without the prefix "Default_". Table is "__EFMigrationHistory" and I'm trying to read the last migration. I am using PostgreSQL 14 for the database.

Whenever the query input in the Orchard UI gets transcribed to actual SQL, it seems I cannot avoid the prefix "Default_" being added to the table name, resulting in an error.

My query is:
```sql
SELECT MigrationId
FROM __EFMigrationHistory
ORDER
	BY MigrationId DESC
	LIMIT 1
```

And the resulting query is:
```sql
SELECT "MigrationId" FROM "Default___EFMigrationHistory" ORDER BY "MigrationId" DESC limit 1;
```

Table obviously does not exist: `An error occurred while executing the SQL query: 42P01: relation "Default___EFMigrationHistory" does not exist POSITION: 27`.

**Is there a way to disable the prefixing or write literal SQL that does not need to be parsed by Orchard?** I could add "Default_" to the table name but that wouldn't be a great idea I have plenty of data.

**Comment by ns8482e** [comment]
Use sql query in c# directly and do not use orchard UI

I guess here  you are trying to query to a table that is not tenant specific and not created by orchard 

The orchard ui query runs in context of current tenant and if current tenant has prefix then all gendered query will add prefix.

**Reply by hishamco** [reply]
There's no official support of EF in OrchardCore, for that you need to handle this in database layer where the migration APIs is there

**Comment by alXDraghiciu** [comment]
For the current implementation, my project uses an API that accesses the database layer, but the client must make a request to fetch the data first.

I was wondering if it could be done in Orchard or C# and then have the data read in a Query in Liquid, to avoid the request.

Any thoughts?

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by hishamco** [reply]
> I was wondering if it could be done in Orchard or C#

In that case no one can help, until you have a more detailed information

**Reply by ns8482e** [reply]
Not sure why you have ___EFMigrationHistory are you using EF for some non orchard specific tables?

**Reply by hishamco** [reply]
I'm sure there's a missing thing in the puzzle here ;)
